sql_metadb/derived_tables/agreements_custom_property.sql

```diff
@@ -1,22 +1,52 @@
+/*
+ * Creates a derived table on agreements custom properties
+ */
 DROP TABLE IF EXISTS folio_derived.agreements_custom_property;
 
--- Creates a derived table on agreements_custom_property and
--- resolves the definition name from erm_agreements_custom_property_definition
 CREATE TABLE folio_derived.agreements_custom_property AS
 SELECT
-    id AS custom_property_id,
-    definition_id AS custom_property_definition_id,
-    cpd.pd_name AS custom_property_definition_name,
-    parent_id AS custom_property_parent_id
+    agreements_custom_property_container.id AS "custom_property_container_id", -- ID that can be linked to the "custom_properties_id" attribute in the "folio_agreements.subscription_agreement" table
+    agreements_custom_property.id AS "custom_property_id",
+    agreements_custom_property.note AS "custom_property_note",
+    agreements_custom_property.public_note AS "custom_property_public_note",
+    agreements_custom_property_definition.pd_id AS "custom_property_definition_uuid",
+    agreements_custom_property_definition.pd_name AS "custom_property_definition_pd_name",
+    agreements_custom_property_definition.pd_type AS "custom_property_definition_pd_type",
+    agreements_custom_property_definition.pd_description AS "custom_property_definition_pd_description",
+    agreements_custom_property_definition.pd_label AS "custom_property_definition_pd_label",
+    agreements_custom_property_integer.id AS "custom_property_integer_id",
+    agreements_custom_property_integer.value AS "custom_property_integer_value",
+    agreements_custom_property_text.id AS "custom_property_text_id",
+    agreements_custom_property_text.value AS "custom_property_text_value"
```

Comment:
If quotation marks ( `"` ) are used around identifiers even (as here) when they are not necessary, be careful that there are no unusual characters accidentally included, because with the quotation marks there Postgres will accept even spaces as part of an identifier name!